build.sbt

```diff
@@ -26,7 +26,11 @@ def commonProject(id: String, projectPath: String, scalacVersion: String): Proje
   }
   val target = s"$id-$versionSuffix"
   Project(target, file(target)).settings(
-    name := id,
+    name := {
+      // Shade only non-test jars since they're most exposed to transitive dependency hell.
+      def majorVersion = version.value.split('.').head
+      if (id.contains("test")) id else s"$id-v$majorVersion"
```

Comment:
Why not just shade every artifact? I don't think it hurts to shade the test helper artifacts... these tend to have a lot of dependencies too.

Reply:
1. That'd have more migration impact.
2. There's less need.
3. The test jars don't share a common package root, so trickier.